core/iwasm/compilation/aot_emit_aot_file.c

```diff
@@ -1633,6 +1638,7 @@ get_name_section_size(AOTCompData *comp_data)
                         read_leb_uint32(p, p_end, func_index);
                         offset = align_uint(offset, 4);
                         EMIT_U32(func_index);
+                        /* TODO: may not necessary to check function indexes */
```

Comment:
I remember the same checks in loader were added by Alibaba developers, I think it should be necessary, at least in loader.